config/initializers/prometheus.rb

```diff
@@ -19,9 +19,15 @@ def start_exporter_server
   # Increase the server thread priority for better latency
   server.instance_variable_get(:@runner).priority = 3
 
-  require './lib/prometheus/graphql_collector'
-
-  server.collector.register_collector(GraphQLCollector.new)
+  if ENV['APPLICATION_TYPE'] == 'compliance' || Rails.env.development? && ENV['HOSTNAME'] == 'rails'
+    require './lib/prometheus/graphql_collector'
+    require './lib/prometheus/business_collector'
+    require './lib/prometheus/engineering_collector'
+
+    server.collector.register_collector(GraphQLCollector.new)
+    server.collector.register_collector(BusinessCollector.new)
+    server.collector.register_collector(EngineeringCollector.new)
```

Comment:
@skateman are we sure that we want to do this? There are X pods running Rails. Having each generate these high-load metrics could lead to service degradation. Also considering that it runs with higher process priority.

Check also https://github.com/RedHatInsights/compliance-backend/pull/1040#issuecomment-945851646

Reply:
I'll revert this